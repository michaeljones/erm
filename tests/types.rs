extern crate erm;
extern crate im;
extern crate insta;
extern crate logos;
extern crate unindent;

mod common;

mod types {

    use common::eval;

    #[test]
    fn main_string_type_signature() {
        let src = r#"
        module Main exposing (..)

        main : List String -> String
        main args =
          "hello, world with type annotation"
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }

    #[test]
    fn other_type_annotation() {
        let src = r#"
        module Main exposing (..)

        add1 : Int -> Int
        add1 x =
          x + 1

        main : List String -> String
        main args =
          String.fromInt (add1 3)
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }
}
