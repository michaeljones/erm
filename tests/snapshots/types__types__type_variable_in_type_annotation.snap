---
source: tests/types.rs
expression: result

---
Hello with type variable
